dial.go

```diff
@@ -93,61 +98,71 @@ func (d *Dialer) Dial(ctx context.Context, raddr ma.Multiaddr, remote peer.ID) (
 			}
 		}()
 
-		maconn, err := d.rawConnDial(ctx, raddr, remote)
+		tptConn, err := d.rawConnDial(ctx, raddr, remote)
 		if err != nil {
 			errOut = err
 			return
 		}
 
 		if d.Protector != nil {
-			pconn, err := d.Protector.Protect(maconn)
+			var pconn tpt.Conn
+			pconn, err = d.Protector.Protect(tptConn)
```

Comment:
So in this case, if using a 'protector', we end up just stream multiplexing over a single QUIC conn, am i reading that correctly?

If that is the case, then we might just want to (for now) disable the use of QUIC and private networks at the same time.